### PR TITLE
solution

### PR DESCRIPTION
**Pull request by Nikoramo**
[DEMO LINK](https://Nikoramo.github.io/react_todo-app-add-and-delete/)

### CODE REVIEW

File: src/components/Header/Header.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  setSearch,
+  handleAddTodo,
+  setError,
+  setErrorMessage,
+}) => {
+  const isActicve = todos.filter(todo => !todo.completed);
+  const handleSubmit = (event: React.SyntheticEvent) => {
+    event.preventDefault();
+
+    if (search.trim() === '') {
```
Comment (OleziO) [inline src/components/Header/Header.tsx:26]: Replace with `!search.trim()`. It will be the same.

**Review by OleziO — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 2]
```diff
@@ -1,24 +1,156 @@
...
+    setTodos(updatedTodo);
+    try {
+      const todoToUpdate = todos.find((todo) => todo.id === id);
+
+      if (todoToUpdate) {
+        await client.patch(`/todos/${id}`, {
+          completed: !todoToUpdate.completed,
+          title: todoToUpdate.title,
+          userId: USER_ID,
+          id,
```
Comment (VitaliyBodnar) [inline src/App.tsx:108]: this code is related to the API part, so please move it to the API file

File: src/styles/animation.scss [hunk 3]
```diff
@@ -0,0 +1,46 @@
+.item-enter {
+  max-height: 0;
+}
+
+.item-enter-active {
+  overflow: hidden;
+  max-height: 58px;
+  transition: max-height 0.3s ease-in-out;
+}
```
Comment (VitaliyBodnar) [inline src/styles/animation.scss:9]: Please follow to the BEM structure 
```suggestion
.item-enter {
  max-height: 0;
  
  &-active {
  overflow: hidden;
  max-height: 58px;
  transition: max-height 0.3s ease-in-out;
 }
}
```

**Review by VitaliyBodnar — CHANGES_REQUESTED** [review]
Well done 🔥
Let's improve your solution 🧑‍💻

**Pull request closed without merging**